### PR TITLE
[core] adding stl io two missing classes

### PR DESCRIPTION
**Pull request by josep-m-carbonell**
Adds  operator << for std::set and std::weak_ptr

### CODE REVIEW

File: kratos/utilities/stl_io.h [hunk 1]
```diff
@@ -39,6 +41,34 @@ std::ostream& operator<<(std::ostream& rOStream, const std::vector<T>& rVec) {
     return rOStream;
 }
 
+template<class T>
+std::ostream& operator <<(std::ostream& rOStream, const std::set<T>& rVec) {
```
Comment (philbucher) [inline kratos/utilities/stl_io.h]:
```suggestion
std::ostream& operator <<(std::ostream& rOStream, const std::set<T>& rSet) {
```

File: kratos/utilities/stl_io.h [hunk 2]
```diff
@@ -39,6 +41,34 @@ std::ostream& operator<<(std::ostream& rOStream, const std::vector<T>& rVec) {
...
+
+  std::size_t vector_size = rVec.size();
```
Comment (philbucher) [inline kratos/utilities/stl_io.h]:
```suggestion
  const std::size_t set_size = rSet.size();
```

**Review by philbucher — COMMENTED** [review]
minor comments, otherwise nice addition, including tests, :+1: 

Btw your test fails

File: kratos/utilities/stl_io.h (hunk 1 above)
Comment (josep-m-carbonell) [reply kratos/utilities/stl_io.h]: Sorry for the failure of the test. I will correct it. I wanted to know if adding this operator here is correct. The code was not compiling with one of this types as variable. Do you know where is shared_ptr defining this operator?

Comment (philbucher) [reply kratos/utilities/stl_io.h]: I think this is the correct place

Unfortunately I don't know where shared_ptr is defining this operator
@roigcarlo ?

**Review by philbucher — DISMISSED** [review]
looks good to me :+1:

**Comment by josep-m-carbonell** [comment]
I do not have a computer now. I will fix tests on Monday. Do not merge. Thank you.

**Comment by josep-m-carbonell** [comment]
Ok, I think is working now, I have added an extra macro to check for strings (now the compared values are shown when tests fail). If appveyor and travis builds are successful let's please re-approve the PR. Thank you!

**Review by ipouplana — APPROVED** [review]

**Review by philbucher — APPROVED** [review]

File: kratos/includes/checks.h [hunk 3]
```diff
@@ -27,6 +27,9 @@
 #define KRATOS_CHECK_EQUAL(a,b) if(!(a == b)) KRATOS_ERROR << "Check failed because " << #a << " is not equal to " << #b
 #define KRATOS_CHECK_NOT_EQUAL(a,b) if(a == b) KRATOS_ERROR << "Check failed because " << #a << " is equal to " << #b
 
+#define KRATOS_CHECK_STRING_EQUAL(a,b) if(a.compare(b) != 0) KRATOS_ERROR << "Check failed because \"" << a << "\" is not equal to \"" << b << "\""
```
Comment (philbucher) [inline kratos/includes/checks.h:30]: just FYI, compare is the same as == for std::strings

Comment (josep-m-carbonell) [reply kratos/includes/checks.h:30]: The problem was the output of KRATOS_ERROR was "#a" instead of "a".

**Pull request merged**